pageserver/src/tenant.rs

```diff
@@ -3229,6 +3230,30 @@ impl Tenant {
     pub fn cached_synthetic_size(&self) -> u64 {
         self.cached_synthetic_tenant_size.load(Ordering::Relaxed)
     }
+
+    /// Flush any in-progress layers, schedule uploads, and wait for uploads to complete.
+    ///
+    /// This function can take a long time: callers should wrap it in a timeout if calling
+    /// from an external API handler.
+    pub async fn flush_remote(&self) -> anyhow::Result<()> {
+        let timelines = self.timelines.lock().unwrap().clone();
+
+        for (timeline_id, timeline) in timelines {
+            tracing::info!(%timeline_id, "Flushing...");
```

Comment:
I dislike these info's, but I guess these transitions are rare enough? We could just have one timeout'd logging per timeline.

Reply:
Yeah, I could see us downgrading these to debug in the future.  Initially I think we're going to be super-interested in what's going on with flushing when we roll out live migrations, and we should only see these for low-rate intentional migrations (callers shouldn't be using flush_ms for evacuating nodes on failure).

I've removed the redundant other "Flushing" log in `upsert_location` though.